assayist/processor/main_analyzer.py

```diff
@@ -0,0 +1,144 @@
+            cversion = '%s-%s' % (build_info['version'], build_info['release'])
+        elif build_type == 'maven':
+            # What we really want is the contents of the "Maven groupId" etc fields.
+            # However they don't seem to be included in the brew response?!
+            # Instead let's parse it out with what is (as best as I can tell) the algorithm.
+            cnamespace, cname = build_info['name'].split('-', 1)
+            cversion = build_info['version'].replace('_', '-')
+            ctype = 'java'
+        elif build_type == 'buildContainer':
+            cnamespace = 'docker-image'
+            cname = build_info['name']
+            cversion = '%s-%s' % (build_info['version'], build_info['release'])
+            ctype = 'image'
+        else:
+            return None, None
+
+        component = self.get_or_create_component(
+            canonical_namespace=cnamespace,
+            canonical_name=cname,
+            canonical_type=ctype)
+        return component, cversion
+
+    def run(self):
+        """Do the actual processing."""
+        build_info = self.read_metadata_file(self.BUILD_FILE)
+        task_info = self.read_metadata_file(self.TASK_FILE)
+
+        build_type = None
+        if task_info:
+            build_type = task_info['method']
+
+        # construct the component
+        component, canonical_version = self._construct_and_save_component(build_type, build_info)
+
+        # construct the SourceLocation
+        source = build_info['source']
+        sourceLocation = self.get_or_create_source_location(source, canonical_version)
+        sourceLocation.component.connect(component)
+
+        # construct the build object
+        build = self.get_or_create_build(build_info['id'], build_type)
+        build.source_location.connect(sourceLocation)
+
+        # record the rpms associated with this build
+        rpms_info = self.read_metadata_file(self.RPM_FILE)
+        for rpm_info in rpms_info:
+            buildroot_id = rpm_info['buildroot_id']
+            rpm = self.get_or_create_rpm_artifact_from_rpm_info(rpm_info)
+            rpm.build.connect(build)
+            self.__map_buildroot_to_artifact(buildroot_id, rpm)
+
+        # record the artifacts
+        archives_info = self.read_metadata_file(self.ARCHIVE_FILE)
+        images_rpm_info = self.read_metadata_file(self.IMAGE_RPM_FILE)
+        for archive_info in archives_info:
+            if archive_info['type'] == 'log':
+                # No one cares about logs
+                continue
+
+            log.debug('Creating build artifact %s', archive_info['id'])
+            aid = archive_info['id']
+            type = archive_info['btype']
+            checksum = archive_info['checksum']
+            filename = archive_info['filename']
+            buildroot_id = archive_info['buildroot_id']
+            # find the nested arch information or set noarch
+            extra = archive_info.get('extra', {})
+            if extra:
```

Comment:
Replace this if statement and definition of extra with:
```python
arch = archive_info.get('extra', {}).get('image', {}).get('arch', 'noarch')
```

Reply:
NACK, that's what it was but apparently in real data (and subsequently in the test data) the archive_info can contain things like:
```
{
...
'extra': None
}
```

Which of course throws an exception if the first get() in a chain correctly returns a None value.

Reply:
I will add a comment to that effect.

Reply:
Makes sense. Thanks.